fossas/fossa-cli
# [ANE-1092] Path Dependency (2/2)

# Overview

This PR is part of the chain, and will not be merged until `CORE` pr is merged!

- Adds path dependency documentation
- Adds path dependency optimization (avoid rescanning if already scanned)
- Adds path dependency nudge (warn users that they have path dependency, and should use experimental flag) 
- Address PR feedback from PR chain 

## Acceptance criteria

As a user, 
- [ ] `fossa-cli` can license scan path dependencies, only when `` flag is present
- [ ] `fossa-cli` nudges to enable path dependencies, if I have path dependency in my project
- [ ] `fossa-cli` does not scan dependency again, if they have already been analyzed
- [ ] `fossa-cli` can be forced to scan dependency again

## Testing plan

- I relied on automated testing.

Load the endpoint from: https://github.com/fossas/FOSSA/pull/11392

1. Create example: (refer to repro case here: https://fossa.atlassian.net/browse/ANE-383)
2. Build `make install-dev`
3. Perform `fossa-dev analyze -e http://localhost:9578 --fossa-api-key <KEY> repro -p path-dep-1 -r 1 --experimental-analyze-path-dependencies`
4. Perform `fossa-dev analyze -e http://localhost:9578 --fossa-api-key <KEY> repro -p path-dep-1 -r 2 --experimental-analyze-path-dependencies` (since path dep is already scanned, you should not see upload/scan msg again)
5. Perform `fossa-dev analyze -e http://localhost:9578 --fossa-api-key <KEY> repro -p path-dep-1 -r 3 --experimental-analyze-path-dependencies --force-vendored-dependency-rescans` (performs upload/scan again)

## Risks

N/A

## References

https://fossa.atlassian.net/browse/ANE-1092

## Checklist
